### ISSUE TITLE
TF 2.0 Beta: custom metric example from documentation is wrong

### ISSUE THREAD

**Issue by kernelizd** [tensorflow/tensorflow#29088]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below):
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I implemented the custom metric shown in this page (CatgoricalTruePositives) https://www.tensorflow.org/alpha/guide/keras/training_and_evaluation

I think it is full of bugs, it has some comments saying (#TODO: fix this). Any way, here what's particularly wrong about it. The accuracy of the NN approaches 99%, yet, this metric says:

binary_true_positives: 8459.0000

(as shown on the website too). If number of samples in MNIST is 50,000, then at least 45k of them should be true positives.  It is unstable. I messed with it once and I got 49k true positive (which makes total sense). Then I reran it and it returned to 8k.

**Describe the expected behavior**
The results are shown on the website. For a low loss of 0.03, the true positives should be close to 50k. However they're shown to be 8k only.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by achandraa** [comment]
Tried to execute the code and getting loss as 0.066 and categorical_true_positive : ~7863.

**Comment by kernelizd** [comment]
Hi. Yes, I got the same thing. Now, if you add one more metric (accuracy), you will notice that the 0.06 loss you got actually correspond to very high accuracy (above 90%), which means that true positives should be well above 40,000. Not 7863. Please notice that we have 50k examples.

**Comment by kernelizd** [comment]
Hi. I have some great news for you. I solved it.
Here what's wrong with your docs:

In the update_state method, you should have this:

>       y_pred = tf.reshape(tf.argmax(y_pred, axis=1), shape=(-1, 1))

Currently it is:

>         y_pred = tf.argmax(y_pred)

With my fix, the categorical true positives will be close to 50k which makes total sense
50000/50000 [==============================] - 1s 29us/sample - loss: 0.0905 - categorical_true_positives: 48647.0000 - accuracy: 0.9729

Because the accuracy is close to 100%.

Explanation: your argmax should act along axis 1 not axis 0 which is the batch axis. It was returning nonsensical stuff like 49 and 64 which are indices of batch datapoints. Second thing, you need to reshape the result to match the shape of y_pred which is what I did.

**Comment by pavithrasv** [comment]
@kernelizd Thank you for looking into this. Do you want to send a PR updating the doc?

**Comment by kernelizd** [comment]
Sure. Will do.

**Comment by kernelizd** [comment]
@pavithrasv 
Hi, I'm not an expert in github. I've just learned the basics. I made the pull request #674 
I hope I did it correctly. Let me know.

**Comment by pavithrasv** [comment]
@kernelizd Thank you for creating a PR, the link you have provided is pointing to a different PR, can you share the entire url may be?

**Comment by kernelizd** [comment]
@pavithrasv 

I'm sorry. Here is the url you asked for:

https://github.com/tensorflow/docs/pull/674

**Comment by pavithrasv** [comment]
Looks like the diff is more than just this line that has changed. You may need to update your PR.

**Comment by kernelizd** [comment]
Hi @pavithrasv  
It is exactly one line. No more.
But for some reason, Github says everything got changed. If you investigate it yourself, you see that there is no change at all.

**Comment by pavithrasv** [comment]
Seems like indentation mismatch.

**Comment by kernelizd** [comment]
I guessed as much. Different Jupyter version?

**Comment by lamberta** [comment]
Fixed in PR
